test/outcome.js

```diff
@@ -0,0 +1,243 @@
+var agent = require('..').start({
+  serviceName: 'test',
+  secretToken: 'test',
+  captureExceptions: true,
+  metricsInterval: 0,
+  centralConfig: false
+})
+const constants = require('../lib/constants')
+
+const noOpClient = {
+  _write () {},
+  sendSpan () {},
+  sendTransaction () {},
+  sendError () {},
+  sendMetricSet () {},
+  flush () {}
+}
+agent._transport = noOpClient
+
+const test = require('tape')
+
+test('span outcome tests', function (test) {
+  test.test('span starts unknown, ends with success', function (t) {
+    // test that
+    agent.startTransaction('foo', 'type', 'subtype', 'action')
+    const span = agent.startSpan()
+    t.equals(span.outcome, constants.OUTCOME_UNKNOWN, 'spans start with unknown outcome')
```

Comment:
Q: I know that we init `span.outcome` to "unknown", but is that a promise we care to enshrine in the tests? The implementation could just as well have been ".outcome is not set (i.e. null or undefined), until span completion, whereupon the .outcome value is determined".

AFAIK unended spans are not included in a trace sent to APM server.

Reply:
Per ~offline~ off GitHub conversation, we'll leave this as is for now, but be open to refactoring the tests if we need to consider a world where spans don't have a default outcome value set.